ext/msgpack/factory_class.c

```diff
@@ -74,14 +75,19 @@ static VALUE Factory_initialize(int argc, VALUE* argv, VALUE self)
 
     fc->has_symbol_ext_type = false;
 
-    switch (argc) {
-    case 0:
-        break;
-    default:
-        // TODO options is not supported yet
-        rb_raise(rb_eArgError, "wrong number of arguments (%d for 0)", argc);
+    VALUE options = Qnil;
+    VALUE strict = Qfalse;
+    rb_scan_args(argc, argv, "0:", &options);
+
+    if (!NIL_P(options)) {
+        static ID keyword_ids[1];
+        if (!keyword_ids[0])
+            keyword_ids[0] = rb_intern("strict");
+        rb_get_kwargs(options, keyword_ids, 0, 1, &strict);
     }
 
+    fc->strict_types = strict;
```

Comment:
You are assigning a `VALUE` into a `bool` no? I think you need `strict_types = RBTEST(strict)`.